release-common-lib/src/main/java/org/reactome/util/compare/DBObjectComparer.java

```diff
@@ -147,225 +176,428 @@ public static int compareInstances(
+			getRegularAttributes(instance);
+
+		return filterAttributes(allAttributes, attributeNameFilter);
+	}
+
+	/**
+	 * Retrieves "regular" (i.e. not referrer) attributes for the passed instance.  Uses a cache for the attributes of
+	 * instances which have been previously queried.
+	 * @param instance Instance for which to get attributes
+	 * @return List of "regular" (i.e. not referrer) SchemaAttributes of the instance passed
+	 */
+	@SuppressWarnings("unchecked")
+	private static List<SchemaAttribute> getRegularAttributes(GKInstance instance)
+	{
+		List<SchemaAttribute> regularAttributes = instanceToRegularAttributesMap.get(instance);
+		if (regularAttributes == null)
+		{
+			regularAttributes = new ArrayList<SchemaAttribute>(instance.getSchemClass().getAttributes());
+			instanceToRegularAttributesMap.put(instance, regularAttributes);
+		}
+		return regularAttributes;
+	}
+
+	/**
+	 * Retrieves referrer attributes for the passed instance (i.e. attributes used by other instances to refer to the
+	 * passed instance).  Uses a cache for the attributes of instances which have been previously queried.
+	 * @param instance Instance for which to get referrer attributes
+	 * @return List of referrer SchemaAttributes of the instance passed
+	 */
+	@SuppressWarnings("unchecked")
+	private static List<SchemaAttribute> getReferrerAttributes(GKInstance instance)
+	{
+		List<SchemaAttribute> referrerAttributes = instanceToReferrerAttributesMap.get(instance);
+		if (referrerAttributes == null)
+		{
+			referrerAttributes = new ArrayList<SchemaAttribute>(instance.getSchemClass().getReferers());
+			instanceToReferrerAttributesMap.put(instance, referrerAttributes);
+		}
+		return referrerAttributes;
+	}
+
+	/**
+	 * Filters the passed list of attributes based on the predicate function passed and returns the filtered list.
+	 * @param attributes List of attributes to filter
+	 * @param attributeNameFilter Predicate value used to determine which values should be included
+	 * @return List of SchemaAttributes after filtering
+	 */
+	private static List<SchemaAttribute> filterAttributes(
+		List<SchemaAttribute> attributes, Predicate<? super SchemaAttribute> attributeNameFilter
+	)
+	{
+		return attributes
+			.stream()
+			.filter(attributeNameFilter != null ? attributeNameFilter : getDefaultAttributeNameFilter())
+			.collect(Collectors.toList());
+	}
+
+	/**
+	 * Returns the default attribute filter as a Predicate function.  By default, the attributes "DB_ID", "dateTime",
+	 * "modified", and "created" are filtered out (i.e. ignored).
+	 * @return Predicate function used to filter out SchemaAttributes value by their names
+	 */
+	private static Predicate<? super SchemaAttribute> getDefaultAttributeNameFilter()
+	{
+		/*
+		 * Used for filtering out attributes that you don't want to compare because they will
+		 * probably create too much noise if you do.
+		 */
+		return a -> {
```

Comment:
A more descriptive name for this variable would be nice

Reply:
Renamed to "attribute"!